runtime/j9vm/jvm.c

```diff
@@ -3968,12 +3969,18 @@ JVM_LoadLibrary(const char *libName, jboolean throwOnFailure)
 			UDATA handle = 0;
 			UDATA flags = J9_ARE_ANY_BITS_SET(javaVM->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_LAZY_SYMBOL_RESOLUTION) ? J9PORT_SLOPEN_LAZY : 0;
 			UDATA slOpenResult = j9sl_open_shared_library((char *)libName, &handle, flags);
-
 			Trc_SC_LoadLibrary_OpenShared(libName);
+			attemptedLoad = TRUE;
+
+/* jdk17+ calls JVM_LoadLibrary with decorated library names. If the following is done
+ * then it overwrites the real error with a failure to load "liblib<name>.so.so".
+ */
```

Comment:
Actually, we don't expect this function to be used at all prior to jdk17; I'll plan to create a follow-up PR that simply asserts that and remove unnecessary code.

Reply:
We can use `defined(OPENJ9_BUILD)` to avoid impacting IBM Java 8 builds.